base/compiler/abstractinterpretation.jl

```diff
@@ -1202,7 +1203,7 @@ function abstract_eval_statement(interp::AbstractInterpreter, @nospecialize(e),
                 t = Const(true) # definitely assigned previously
             end
         elseif isa(sym, Symbol)
-            if isdefined(sv.mod, sym.name)
```

Comment:
This seems to be an obvious bug, but maybe the current abstract interpretation won't touch this pass and so any bug from this happens so far ? Well I believe this is better to be fixed anyway.

Reply:
I don't think we allow raw Symbols anymore to refer to globals. Only `GlobalRef`. Should probably delete branch entirely. @JeffBezanson?

Reply:
We do still use this form, but we always convert it to `GlobalRef` if we're going to call inference.